### PR TITLE
1856: Add step for CGR merger to remove down to 10 tokens

### PR DESCRIPTION
**Pull request by ventusignis**

### CODE REVIEW

File: lib/engine/game/g_1856/game.rb [hunk 1]
```diff
@@ -1598,12 +1601,17 @@ def national_token_swap
 
           # Then reduce down to limit
           # TODO: Possibly override ReduceTokens?
-          if national.tokens.size > national_token_limit
+          tokens_to_keep = [home_bases.size, national_token_limit].max
+          if national.tokens.size > tokens_to_keep
             @log << "#{national.name} will is above token limit and must decide which tokens to remove"
```
Comment (roseundy) [inline lib/engine/game/g_1856/game.rb:1606]: Delete the word "will"

File: lib/engine/game/g_1856/game.rb [hunk 2]
```diff
@@ -1598,12 +1601,17 @@ def national_token_swap
...
-            # TODO: implement this case, maybe use a varaiation of the below?
-            # @round.corporations_removing_tokens = [buyer, acquired_corp]
+            # This will be resolved in RemoveTokens
+            @round.pending_removals << {
+              corp: national,
+              count: national.tokens.size - tokens_to_keep,
+              hexes: national.tokens.reject { |t| home_bases.include?(t.hex) }.map(&:hex),
```
Comment (tobymao) [inline lib/engine/game/g_1856/game.rb:1611]: tokens.map(&:hex).reject { |hex| home_bases.include?(hex) }

File: lib/engine/game/g_1856/game.rb (hunk 1 above)
Comment (ventusignis) [reply lib/engine/game/g_1856/game.rb:1606]: Fixed

File: lib/engine/game/g_1856/game.rb (hunk 2 above)
Comment (ventusignis) [reply lib/engine/game/g_1856/game.rb]: This is updated now

**Review by roseundy — APPROVED** [review]

**Pull request merged**